src/nep.cpp

```diff
@@ -2776,6 +2780,35 @@ void NEP3::init_from_file(const std::string& potential_filename, const bool is_r
   }
 }
 
+void NEP3::update_type_map(
+  const int ntype,
+  int* type_map,
+  char** elements)
+{
+  int n = 0;
+  for (int itype = 0; itype < ntype + 1; ++itype) {
+    // check if set NULL in lammps input file
+    if (type_map[itype] == -1) {
+      continue;
+    }
+
+    // find the same element name in potential file
+    std::string element_name = elements[type_map[itype]];
+    for (n = 0; n < paramb.num_types; ++n) {
+      if (element_name == element_list[n]) {
+        type_map[itype] = n;
+        break;
+      }
+    }
+
+    // check if no corresponding element
+    if (n == paramb.num_types) {
+      std::cout << "There is no element " << element_name << " in the potential file." << std::endl;
```

Comment:
Is this function only called by a single MPI thread?

Reply:
I mean it is better to confine the error reporting to a single thread using `if (is_rank_0)`.

Reply:
Yep, you are right, it's better. I check the previous error reporting method and find there is no `if (is_rank_0)`.
![image](https://github.com/user-attachments/assets/58b36b36-9e7f-45ea-bf88-626bd91d59ba)

Reply:
But why we don't use LAMMPS_NS::error to report? (make LAMMPS to consider the MPI hhhh)

Reply:
We cannot use LAMMPS stuff in `nep.cpp` as it is also used for other packages such as `calorine`.

Reply:
It is ok to report error for all threads. This is not a very important thing.